android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -1061,7 +1066,18 @@ public static void openTransaction(TransactionInfo txInfo, JsonRpcService jsonRp
             if (txInfo.txHash == null || txInfo.txHash.isEmpty()) {
                 return;
             }
-            openAddress("/tx/" + txInfo.txHash, jsonRpcService, activity, coinType);
+            TxData1559 tx1559Data =
+                    txInfo.txDataUnion != null ? txInfo.txDataUnion.getEthTxData1559() : null;
```

Comment:
we can't access union members directly, they could be incorrect, we need to check what does it contain first via `which` and access it only if it contains the needed object. Check that as an example https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/util/ParsedTransaction.java#L143